src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -605,6 +618,16 @@ public FormValidation doCheckClusterName(
       return FormValidation.ok();
     }
 
+    public FormValidation doCheckNamespace(@QueryParameter("namespace") final String namespace) {
+      /* Regex from
+       * https://github.com/kubernetes/apimachinery/blob/7d08eb7a76fdbc79f7bc1b5fb061ae44f3324bfa/pkg/util/validation/validation.go#L110
+       */
+      if (!namespace.isEmpty() && !namespace.matches("[a-z0-9]([-a-z0-9]*[a-z0-9])?")) {
```

Comment:
Prefer to check for null similar to the other verify methods and return an error if missing. Throwing a NPE will lead to the devil screen.